arrow-core-data/src/main/kotlin/arrow/core/Validated.kt

```diff
@@ -179,9 +208,39 @@ typealias Invalid<E> = Validated.Invalid<E>
  * //sampleEnd
  * ```
  *
- * Kotlin says that our match is not exhaustive and we have to add `else`.
+ * ### Improving the validation
+ *
+ * Kotlin says that our match is not exhaustive and we have to add `else`. To solve this we need to nest our when,
+ * but that would complicate the code. Alternatively, we could use Applicative through the `nelApplicative` function in arrow-core
+ * to unlock `tupledN`. This function combines Validated by accumulating errors in a tuple, that we can then map, the above function
+ * is then equivalent to this:
+ *
+ * ```kotlin:ank:silent
+ * import arrow.core.NonEmptyList
+ * import arrow.core.Validated
+ * import arrow.core.ValidatedPartialOf
+ * import arrow.core.extensions.nonemptylist.semigroup.semigroup
+ * import arrow.core.extensions.validated.applicative.applicative
+ * import arrow.core.fix
+ * import arrow.typeclasses.Applicative
+ * // added manually due to deps
```

Comment:
Additional white line?

Reply:
I'm not sure I understand why this is necessary? Can we not use `Validated.applicative(NonEmptyList.semigroup<String>()).tupledN` in the ank snippet?

Reply:
Yes, we're actually using that already, I just wanted a quick way of obtaining it for any E